e2e/testmain_test.go

```diff
@@ -74,10 +80,25 @@ func TestMain(m *testing.M) {
 	parseFlagsOrDie()
 
 	ctx := context.Background()
-	client, err := google.DefaultClient(ctx, compute.ComputeScope)
+
+	credentials, err := google.FindDefaultCredentials(ctx, compute.ComputeScope)
 	if err != nil {
 		log.Fatal(err)
 	}
+	ts := credentials.TokenSource
+
+	if testFlags.serviceAccountName != "" {
+		ts, err = impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: testFlags.serviceAccountName,
+			Scopes:          []string{compute.ComputeScope, compute.CloudPlatformScope},
+			Lifetime:        1200 * time.Second,
```

Comment:
20minutes is this enough? Maybe we should parametrize this?

Reply:
it should refresh after this time elapses. I will verify tomorrow

Reply:
If `Lifetime` is not specified, it will auto-refresh (every 1 hour).